projects/npm-tools/packages/npm-scripts/package.json

```diff
@@ -22,41 +22,41 @@
 		"@storybook/addon-viewport": "^5.1.9",
 		"@storybook/core": "5.3.3",
 		"@storybook/react": "^5.1.9",
-		"@testing-library/dom": "5.6.1",
-		"@testing-library/jest-dom": "4.0.0",
-		"@testing-library/react": "8.0.7",
-		"@testing-library/react-hooks": "3.4.1",
-		"@testing-library/user-event": "4.2.4",
+		"@testing-library/dom": "^5.6.1",
+		"@testing-library/jest-dom": "^4.2.4",
+		"@testing-library/react": "^8.0.9",
+		"@testing-library/react-hooks": "^3.4.2",
+		"@testing-library/user-event": "^4.2.4",
 		"babel-eslint": "^10.0.3",
 		"babel-jest": "^25.1.0",
-		"babel-loader": "8.0.6",
+		"babel-loader": "^8.1.0",
 		"babel-plugin-transform-react-remove-prop-types": "0.4.24",
-		"cosmiconfig": "^6.0.0",
+		"cosmiconfig": "^7.0.0",
 		"cross-env": "^7.0.0",
-		"cross-spawn": "7.0.1",
+		"cross-spawn": "^7.0.3",
 		"css-loader": "^3.0.0",
 		"deepmerge": "^4.0.0",
 		"eslint": "7.10.0",
-		"fs-extra": "^8.1.0",
-		"globby": "11.0.0",
+		"fs-extra": "^9.0.1",
+		"globby": "^11.0.0",
 		"gulp": "^4.0.2",
-		"html-webpack-plugin": "4.3.0",
-		"http-proxy-middleware": "0.21.0",
+		"html-webpack-plugin": "^4.3.0",
+		"http-proxy-middleware": "^1.0.5",
 		"jest": "^25.1.0",
 		"jest-environment-jsdom-thirteen": "1.0.1",
-		"jest-fetch-mock": "3.0.1",
+		"jest-fetch-mock": "^3.0.3",
 		"liferay-lang-key-dev-loader": "^1.0.3",
 		"liferay-npm-bridge-generator": "2.19.3",
 		"liferay-npm-bundler": "2.19.3",
 		"liferay-theme-tasks": "10.0.2",
```

Comment:
Interesting that these ones are pinned... what's our plan moving onto the monorepo?

Reply:
I will document this somewhere, but my thinking is:

- Projects with a good track record of following semver and which are >= 1.0.0: Use `^` ranges to make updating easier.
- Projects which are reckless with semver or < 1.0.0: Use `--exact` versions.
- Projects which are exactly 1.0.0: behavior and policy is undefined 😉 
- Our own internal projects: We can exactly lock these like we're already doing because we control the releases (ie. don't have to worry about duplicates), and because it allows us to do a controlled roll-out to liferay-portal (ie. if I `rm yarn.lock` in liferay-portal, I shouldn't suddenly get a new version of the bundler when I run `yarn install` — the only have to update our dependencies should be by updating `@liferay/npm-scripts`.

So Prettier is an interesting example here. We have traditionally locked it because we didn't want the `rm yarn.lock && yarn` scenario to pull in a new version and potentially churn a bunch of files; but at this point in Prettier's lifecycle, it is very stable and moving slowly, and they've shown that they respect semver, so there's no real reason to treat it differently than any other dependency.

Reply:
Created https://github.com/liferay/liferay-frontend-projects/issues/106 to track documenting this, and to give us a place to discuss the details further if we need to.